### DISCUSSION TITLE
Unable to Flash Beca BAC-6000 or BHT-6000

### DISCUSSION THREAD

**Discussion by mojtabavi** [arendst/Tasmota#17246] [Q&A] [answered]
Hello fellow Tasmotians.
I recently purchased a "Smart WIFI Thermostat"-Beca BAC-6000 for my Fancoil
I did search around to see if anyone had a go at it so far with their own firmware, but I couldn't find any info on it. except one
Opened it up, and it is based on the TYWE3S with a SONiX SN8F57084 MCU,
Finally, after doing a lot of research and searching, I found MCU command to Control Thermostat.
And, I write a firmware for it, but the problem is that the size of the firmware is almost 1mb because of webUI and it is not possible to flash the device using OTA or Tuya Convert.
So I decided to flash ESP using the CH340G programmer module, but there is a problem, the ESP module is not boot in the programing mode and sends incomprehensible characters on the serial monitor.
![Screenshot 2022-12-01 141155](https://user-images.githubusercontent.com/63472358/205396653-f1d68caa-1dfe-4e87-a875-603f279793de.png)

Connection between programmer and ESP:
esp-gnd --> gnd
esp-3v3 --> 3v3
esp-gpio0 --> gnd
esp-rx --> tx
esp-tx --> rx
![photo_2022-12-01_14-09-15](https://user-images.githubusercontent.com/63472358/205396864-a7a94670-d02f-48bc-a709-abac42f322b3.jpg)

Is the problem due to the disturbance of the MCU signal on the common bus?
Or is the problem due to the lack of power supply of the programmer?
I will be very happy if we can exchange ideas here and find the problem.

**Comment by barbudor** [comment]
Why reposting?
You already have a topic open with unanswered questions

**Comment by mojtabavi** [comment, marked as answer]
Surprisingly, by changing the programmer instead of using CH340G, the problem was solved

**Comment by wrongisthenewright** [comment]
> Hello fellow Tasmotians. I recently purchased a "Smart WIFI Thermostat"-Beca BAC-6000 for my Fancoil I did search around to see if anyone had a go at it so far with their own firmware, but I couldn't find any info on it. except one Opened it up, and it is based on the TYWE3S with a SONiX SN8F57084 MCU, Finally, after doing a lot of research and searching, I found MCU command to Control Thermostat. 

Hi, I have a similar thermostat(BAC-005), it has the same chip, but was unable to find a functioning firmware, can you share your solution?
